dotCMS/src/main/java/com/dotcms/rest/api/v1/contenttype/ContentTypeResource.java

```diff
@@ -78,23 +75,28 @@ public ContentTypeResource(final ContentTypeHelper contentletHelper, final WebRe
 	@NoCache
 	@Consumes(MediaType.APPLICATION_JSON)
 	@Produces({MediaType.APPLICATION_JSON, "application/javascript"})
-	public final Response createType(@Context final HttpServletRequest req, final String json)
+	public final Response createType(@Context final HttpServletRequest req, final ContentTypeForm form)
 			throws DotDataException, DotSecurityException {
 		final InitDataObject initData = this.webResource.init(null, true, req, true, null);
 		final User user = initData.getUser();
 
 		Response response = null;
 
 		try {
-			final List<ContentType> typesToSave = new JsonContentTypeTransformer(json).asList();
+			Iterable<ContentTypeForm.ContentTypeFormEntry> typesToSave = form.getIterable();
 			final List<ContentType> retTypes = new ArrayList<>();
 
 			// Validate input
-			for (ContentType type : typesToSave) {
+			for (ContentTypeForm.ContentTypeFormEntry entry : typesToSave) {
+				ContentType type = entry.contentType;
+				List<String> workflowsIds = entry.workflowsIds;
+
 				if (UtilMethods.isSet(type.id()) && !UUIDUtil.isUUID(type.id())) {
 					return ExceptionMapperUtil.createResponse(null, "ContentType 'id' if set, should be a uuid");
```

Comment:
add some log here

Reply:
Also should provide an accurate status, such as 422

Reply:
On the error handling should be aware about permissions, and return an error and status accord to it